modules/puppetca_token_whitelisting/puppetca_token_whitelisting_autosigner.rb

```diff
@@ -0,0 +1,96 @@
+require 'jwt'
+require 'openssl'
+
+module ::Proxy::PuppetCa::TokenWhitelisting
+  class Autosigner
+    include ::Proxy::Log
+    include ::Proxy::Util
+
+    def sign_all
+      Proxy::PuppetCa::TokenWhitelisting::Plugin.settings.sign_all
+    end
+
+    def hosts_file
+      Proxy::PuppetCa::TokenWhitelisting::Plugin.settings.hosts_file
+    end
+
+    def smartproxy_cert
+      OpenSSL::PKey::RSA.new File.read Proxy::SETTINGS.ssl_private_key.to_s
```

Comment:
What happens if the smart proxy is run without ssl?

Reply:
Well, that would be a problem. I guess we would have to ensure that there is a RSA certificate there if used for SSL or not...